client/web/src/enterprise/campaigns/settings/backend.ts

```diff
@@ -61,13 +63,14 @@ export function deleteCampaignsCredential(id: Scalars['ID']): Promise<void> {
 }
 
 export const queryUserCampaignsCodeHosts = ({
+    username,
     first,
     after,
 }: Partial<UserCampaignsCodeHostsVariables>): Observable<CampaignsCodeHostsFields> =>
     requestGraphQL<UserCampaignsCodeHostsResult, UserCampaignsCodeHostsVariables>(
         gql`
-            query UserCampaignsCodeHosts($first: Int, $after: String) {
-                currentUser {
+            query UserCampaignsCodeHosts($username: String!, $first: Int, $after: String) {
+                user(username: $username) {
                     campaignsCodeHosts(first: $first, after: $after) {
                         ...CampaignsCodeHostsFields
                     }
```

Comment:
If you use
```suggestion
                node(id: $userID) {
                ... on User {
                    campaignsCodeHosts(first: $first, after: $after) {
                        ...CampaignsCodeHostsFields
                    }
                    }
```
then you would not need to pass in both userID AND username in the component using it